j5/backends/hardware/sr/v4/serial/protocol.py

```diff
@@ -58,17 +59,33 @@ def firmware_version(self) -> str:
         identity = self.get_identity()
         return identity.software_version
 
-    def check_firmware_version_supported(self) -> None:
+    def _check_firmware_version_supported(
+        self,
+        version: str,
+        minimum_minor_version: int = 4,
+    ) -> None:
         """
         Raises an exception if the firmware version is not supported.
 
-        :raises NotImplementedError: power board is running unsupported firmware
+        :param version: the firmware version string to check.
+        :param minimum_minor_version: the lowest supported minor firmware version
+        :raises CommunicationError: the board is running unsupported firmware
         """
-        version = self.firmware_version
-        if not version.startswith("4."):
-            raise NotImplementedError(f"This power board is running firmware "
-                                      f"version {version},"
-                                      "but only version 4.x is supported.")
+        match = re.match(r"^(\d+)\.(\d+)$", version)
+        if not match:
+            raise CommunicationError(f"Unable to parse version number: {version}")
+
+        major, minor = match.groups()
```

Comment:
The first digit here is the hardware version, the second segment is the major version.

Reply:
This would be useful to have documented in the protocol docs

Reply:
0a20492